aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/contexts/LambdaSegmentContext.java

```diff
@@ -37,7 +37,9 @@
     private static final String LAMBDA_TRACE_HEADER_KEY = "_X_AMZN_TRACE_ID";
 
     private static TraceHeader getTraceHeaderFromEnvironment() {
-        return TraceHeader.fromString(System.getenv(LAMBDA_TRACE_HEADER_KEY));
+        return TraceHeader.fromString(System.getenv(LAMBDA_TRACE_HEADER_KEY) != null 
+            ? System.getenv(LAMBDA_TRACE_HEADER_KEY) 
+            : System.getProperty(LAMBDA_TRACE_HEADER_KEY));
```

Comment:
Perhaps we should introduce a string that's formatted as a more idiomatic system property key, e.g. `com.amazonaws.xray.traceHeader`

Reply:
And also if you could link #251 in a comment in the code so we know why this behavior is different for our java SDK than in other languages.

Reply:
I can definitely do that, but I'm not clear on what you are asking for?  Are you asking for this?

private static final String LAMBDA_TRACE_HEADER_PROP = "com.amazonaws.xray.traceHeader";
 return TraceHeader.fromString(System.getenv(LAMBDA_TRACE_HEADER_KEY) != null 
            ? System.getenv(LAMBDA_TRACE_HEADER_KEY) 
            : System.getProperty(LAMBDA_TRACE_HEADER_PROP));

Reply:
Yep exactly like that!